SIPS/expand-fee-reclamation.md

```diff
@@ -0,0 +1,69 @@
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SIP.-->
+
+Expand fee reclamation to use next price update for certain synths
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+Add an optional mechanism to fee reclamation where instead of a waiting period in time for calculating price differences, some synths would require a new price update.
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+The underlying assets of certain synths - such as forex, commodities, stocks and indicies - undergo spot market closures. For forex and commodities, spot market closures are weekends across all world timezones (Friday at 5 pm EST - Sunday 5 pm EST). For stocks and indicies, there are evenings, weekends and public holidays for in each market's region. During these periods where spot markets are closed, the futures markets continue to operate and world news continues to be delivered that may impact futures and shift the spot price at next market open. In order to prevent Synthetix traders taking advantage of these upcoming movements, we propose expanding fee reclamation for these synths, so that instead of waiting some number of time in the future, fee reclamation will wait until a new price has been received.
+
+Moreover, this functionality would be useful for leveraged synths. If on-chain prices are only updated every hour (and potentially variant per pricing network) and on a 1% deviation via Chainlink's decentralized pricing networks, leveraged synths with only a timed fee reclamation waiting period could expose the debt pool to significant frontrunning opportunites. Instead, this proposal would allow adding leveraged synths into this
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+
+1. `Exchanger` would need a new `mapping(bytes32 => bool)` to track which synths require this new "next price" waiting period (others would continue to use the timed waiting period)
+2. `Exchanger.maxSecsLeftInWaitingPeriod()` would always return `0` for those using the "next price" feature
+3. `ExchangeRates` needs new function to `getNextRoundId(fromRoundId, bytes32)`
+4. `Synthetix.isWaitingPeriod(bytes32)` would need to be amended, if any exchange into (`dest`), out of (`src`) (or both) a "next price" synth, then needs to lookup in `ExchangeRates.getNextRoundId()` to see if a new price has been received or not (for one or both)
```

Comment:
`ExchangeRates.getNextRoundId()` for Forex / Commodities will get a new roundID as the Chainlink heartbeat currently will update new roundID every hour.  

It'll require them to deprecate that if we rely on the next price / roundID to implement market closure.

Reply:
Correct - either we need to check to see if the price hasn't moved or that the Chainlink oracles provide a marketClosed `bool` we can read (which is the preference)